lib/infer/membership.js

```diff
@@ -77,6 +116,32 @@ function countModuleIdentifiers(comment, identifiers) {
 }
 
 /**
+ * Returns the comment object after normalizing Foo.prototype and Foo# expressions
+ * @param {Object} comment parsed comment
+ * @returns {Object} the normalized comment
+ */
+function normalizeMemberof(comment) {
+  var memberof = comment.memberof;
+  var index = memberof.lastIndexOf('.');
+
+  if (index !== -1 && memberof.slice(index + 1) === 'prototype') {
```

Comment:
Probably the most compact way to express this would be:

```js
  var isPrototype = /.prototype$/;
  if (memberof.match(isPrototype)) {
    comment.memberof = memberof.replace(isPrototype, '');
    comment.scope = 'instance';
    return comment;
  }
```

Performance certainly matters, but with tiny strings in this place I don't think even a wasteful implementation could take more than 1% of total time.